kore/test/Test/Kore/Builtin/List.hs

```diff
@@ -258,6 +270,106 @@ test_concatAssociates =
         (===) evalConcat12_3 evalConcat1_23
         (===) Pattern.top =<< evaluateT predicate
 
+test_concatSymbolic :: TestTree
+test_concatSymbolic =
+    testPropertyWithSolver
+        "concat{}(element{}(x), xs) === concat{}(element{}(y), ys))\n\
+        \concat{}(xs, element{}(x)) === concat{}(ys, element{}(y))"
+        prop
+  where
+    prop = do
+        let elemVarX = "x" `ofSort` intSort
+            patSymbolicX = mkElemVar elemVarX
+            patSymbolicY = mkElemVar $ "y" `ofSort` intSort
+            elemVarXs = "xs" `ofSort` listSort
+            patSymbolicXs = mkElemVar elemVarXs
+            patSymbolicYs = mkElemVar $ "ys" `ofSort` listSort
+            patElemX =
+                List.internalize testMetadataTools $ elementList patSymbolicX
+            patElemY =
+                List.internalize testMetadataTools $ elementList patSymbolicY
+
+            patConcatX = concatList patElemX patSymbolicXs
+            patConcatY = concatList patElemY patSymbolicYs
+            patUnifiedXY = mkAnd patConcatX patConcatY
+
+            patConcatX' = concatList patSymbolicXs patElemX
+            patConcatY' = concatList patSymbolicYs patElemY
+            patUnifiedXY' = mkAnd patConcatX' patConcatY'
+            expect = Conditional
+                        { term = patConcatY
+                        , predicate = makeTruePredicate listSort
+                        , substitution =
+                            from $ Map.fromList
+                                [ (ElemVar elemVarX, patSymbolicY)
+                                , (ElemVar elemVarXs, patSymbolicYs)
+                                ]
+                        }
+            expect' = Conditional
+                        { term = patConcatY'
+                        , predicate = makeTruePredicate listSort
+                        , substitution =
+                            from $ Map.fromList
+                                [ (ElemVar elemVarX, patSymbolicY)
+                                , (ElemVar elemVarXs, patSymbolicYs)
+                                ]
+                        }
+        unified <- evaluateT patUnifiedXY
+        unified' <- evaluateT patUnifiedXY'
+        expect === unified
+        expect' === unified'
```

Comment:
I think these tests would be easier to understand if the `expect` was adjacent to the assertion itself.
```suggestion
            patConcatX = concatList patElemX patSymbolicXs
            patConcatY = concatList patElemY patSymbolicYs
            patUnifiedXY = mkAnd patConcatX patConcatY

            expect = Conditional
                        { term = patConcatY
                        , predicate = makeTruePredicate listSort
                        , substitution =
                            from $ Map.fromList
                                [ (ElemVar elemVarX, patSymbolicY)
                                , (ElemVar elemVarXs, patSymbolicYs)
                                ]
                        }
                        
        unified <- evaluateT patUnifiedXY
        expect === unified

        let
            patConcatX' = concatList patSymbolicXs patElemX
            patConcatY' = concatList patSymbolicYs patElemY
            patUnifiedXY' = mkAnd patConcatX' patConcatY'

            expect' = Conditional
                        { term = patConcatY'
                        , predicate = makeTruePredicate listSort
                        , substitution =
                            from $ Map.fromList
                                [ (ElemVar elemVarX, patSymbolicY)
                                , (ElemVar elemVarXs, patSymbolicYs)
                                ]
                        }

        unified' <- evaluateT patUnifiedXY'
        expect' === unified'
```